lib/site_setting_validations.rb

```diff
@@ -5,6 +5,10 @@ def validate_error(key)
     raise Discourse::InvalidParameters.new(I18n.t("errors.site_settings.#{key}"))
   end
 
+  def validate_censored_words(new_val)
+    validate_error :invalid_character if new_val.match(/^\w*$/).nil?
```

Comment:
Looks like no one wrote one yet :P

Perhaps start a new spec in `site_setting_validations_spec.rb`?